test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -6794,7 +6809,7 @@ BOOST_AUTO_TEST_CASE(function_types_sig)
 			}
 		}
 	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
+	CHECK_WARNING(text, "Use of var keyword is deprecated");
```

Comment:
`function pure external g = this.h` should work

Reply:
Looks like the next block below tests this exactly, so I'll just remove this block.